apis/core/v1beta1/openstackcontrolplane_types.go

```diff
@@ -188,6 +188,13 @@ type OpenStackControlPlaneSpec struct {
 	// template Section, the globally defined ExtraMounts are ignored and
 	// overridden for the operator which has this section already.
 	ExtraMounts []OpenStackExtraVolMounts `json:"extraMounts,omitempty"`
+
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:default=false
+	// RandomPodNames - Use a unique prefix for cinder and Glance CRs to have unique pod names.
+	// Convenient to avoid podname (and thus hostname) collision between different deployments.
+	// Useful for CI jobs as well as preproduction and production environments that use the same storage backend, etc.
+	RandomPodNames bool `json:"randomPodNames"`
```

Comment:
I would put this into the cinder and the glance section of the Spec respectively that would:
* signal that this is only applied to the give subresource
* allow to enable the feature selectively per subresource